src/EncounterPage/HPCalculator.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react'
+import CalculatorButton from './CalculatorButton'
+import styled from 'styled-components'
+
+export default function HPCalculator({ setCurrentHP, currentHP, maxHP }) {
+  const [hpInputValue, setHpInputValue] = useState('')
+  return (
+    <CounterStyled>
+      <CalculatorButton onClick={addToHP} operator="+" />
+      <InputStyled
+        min={0}
+        value={hpInputValue}
+        onChange={(event) => handleChange(event)}
+        name="HPInput"
+        type="tel"
```

Comment:
You used type="tel". That normally means it is a telephone number.  I think type=number would be more useful. If you want to limit fractions you can have a look at this link:
https://stackoverflow.com/questions/11302360/input-type-for-html-form-for-integer